tests/test_hier_config.py

```diff
@@ -86,8 +86,16 @@ def test_dump_and_load_from_dump_and_compare(self):
 
         hier_post_dump = HConfig(self.host_a, self.os, self.options)
         hier_post_dump.load_from_dump(dump)
+        # py2.7 does not eval an object the same way
```

Comment:
What do you mean by this? https://github.com/netdevops/hier_config/blob/master/hier_config/hc_child.py#L70 Implements a custom __eq__ method to handle this case specifically.

Reply:
https://github.com/netdevops/hier_config/blob/ba00aa2063c6883544679b652a28e687d21226bf/hier_config/hc_child.py#L88

I believe the issue is that you only implemented a custom __eq__ method and not the other comparison methods, particularly __ne__ for this example.

Reply:
perhaps a  quick fix would be `if not self_child == other_child:`

Reply:
Per https://stackoverflow.com/questions/4352244/python-should-i-implement-ne-operator-based-on-eq Python3 automatically understands the relationship between `==` and `!=`. Sounds like the following should be added if we wish to facilitate Python2.7 support:
```Python
def __ne__(self, other):
    return not self.__eq__(other)
```